pkg/units/control.go

```diff
@@ -25,6 +25,13 @@ import (
 	"fmt"
 )
 
+type filter struct {
```

Comment:
Add a comment explaining what is this and why is it here.

Also Does it need to live in the control.go file? isn't it better for it to be located in pkg/cbpf?